src/actions/classifier.js

```diff
@@ -35,3 +51,143 @@ export function fetchWorkflow(workflowID) {
    })
   }
 }
+
+export function fetchTutorials(workflowID) {
+  return dispatch => {
+    return new Promise ((resolve) => {
+      apiClient.type('tutorials').get({workflow_id: workflowID}).then(([tutorial]) => {
+        const tutorialResource = tutorial
+        dispatch(setState(`classifier.tutorial.${workflowID}`, tutorialResource))
+        if (!isEmpty(tutorial)) {
+          let mediaByID = {}
+          tutorialResource.get('attached_images').then((mediaResources) => {
+            forEach((mediaResource) => mediaByID[mediaResource.id] = mediaResource, mediaResources)
+            dispatch(setState(`classifier.tutorial.${workflowID}.mediaResources`, mediaByID))
+            return resolve()
+          }).catch(() => {
+            return resolve()
+          })
+        } else {
+          return resolve()
+        }
+      }).catch(() => { //does not exist for this project, that is OK
+        dispatch(setState(`classifier.tutorial.${workflowID}`, {}))
+        return resolve()
+      })
+    })
+  }
+}
+
+export function fetchProject(workflowID) {
+  return (dispatch, getState) => {
+    const workflow = getState().classifier.workflow[workflowID]
+    const projectID = workflow.links.project
+    return new Promise ((resolve, reject) => {
+      apiClient.type('projects').get({id: projectID}).then(([project]) => {
+        dispatch(setState(`classifier.project.${workflowID}`, project))
+        return resolve()
+      }).catch(() => {
+        return reject()
+      })
+   })
+  }
+}
+
+export function setupProjectPreferences(workflowID) {
+  return (dispatch, getState) => {
+    const workflow = getState().classifier.workflow[workflowID]
+    const projectID = workflow.links.project
+
+    return new Promise ((resolve, reject) => {
+      if (getState().user.isGuestUser){
+        return resolve()
+      }
+
+      dispatch(getAuthUser()).then((userResource) => {
+        return userResource
+      }).then((userResource)=> {
+        userResource.get('project_preferences', {project_id: projectID}).then (([projectPreferences]) => {
+          //Before being able to classify on a project, the user needs to have their preference created if it doesn't exist
+          if (projectPreferences) {
+            return resolve()
+          }
+
+          const projectPreference = {
+            links: { project: projectID },
+            preferences: {}
+          }
+          apiClient.type('project_preferences').create(projectPreference).save().then((projectPreferenceResource) => {
+            return projectPreferenceResource
+          }).then((projectPreferenceResource) => {
+            projectPreferenceResource.get('project').then((project) => {
```

Comment:
I think you were right - they're both for the current workflow's project so it works to remove it and I just get the display name and slug from the stored project instead.